scripts/front/__tests__/add-missing-keys-to-other-language.test.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const fs = require('fs-extra');
```

Comment:
I think we should mock this so we are not actually writing files.

Reply:
@soupette I intentionally want to write files to disk, after the test finishes, I revert it as before the test executes. I always try to make the test as real as possible. 
For the problem of infinite loop (jest watch), I found a solution by configuring [watchPathIgnorePatterns](https://jestjs.io/docs/configuration#watchpathignorepatterns-arraystring) to ignore some test data files (`en.json`, `vi.json`). I think it's safe to ignore all json files, I don't think there will be a test that actually wants to watch json file. If there is, we can always update `watchPathIgnorePatterns`
```diff
// jest.base-config.js
+    watchPathIgnorePatterns: ['<rootDir>/.*\\.json'],
```


`yarn test:unit --watch scripts/front/__tests__/add-missing-keys-to-other-language.test.js`  before update the config (https://github.com/strapi/strapi/pull/12812/commits/d85a387b24e40e63c86cf03ae4ad486704de57e1)
![Mar-15-2022 15-25-37](https://user-images.githubusercontent.com/8603085/158336427-cfb08b8b-be83-44cb-81c9-9d61539a595e.gif)
`yarn test:unit --watch scripts/front/__tests__/add-missing-keys-to-other-language.test.js`  after update the config:
![Mar-15-2022 15-23-23](https://user-images.githubusercontent.com/8603085/158336042-52287f28-f9f5-4d17-bd18-dcb809137862.gif)

Reply:
Sorry but I disagree tests should not write files to disk want we want to test is your algorithm not that the fs lib is working correctly.

Reply:
@soupette What I implemented here is a kind of e2e test. Just want to ignore all the implementation details and test if it works in a real-world example.
However, I completely understand that maybe by the convention of Strapi team, we expect a unit test on the algorithm i this scenario. Or we want tests to run faster by eliminating the File I/O operations here. I can look into that and update the test not to write files to disk later and mention you.
Thanks for your feedback

Reply:
@soupette I just updated to extract the algorithm to a separated function `updateMissingKeysToJSON` and test it. There is no file writing in that function. Can you please take a look? Thanks.